cove.input: Randomise the original file filename

Bjwebb opened the pull request:
https://github.com/ThreeSixtyGiving/standard/issues/353

Before giving the file URL to the user, we will check that they are
authenticated. This change makes sure that the URL can't be guessed.

michaelwood commented on cove/input/models.py:
```diff
@@ -16,8 +18,16 @@
+def upload_to(instance, filename):
+    file_extension = filename.split('.')[-1]
+    if file_extension not in CONTENT_TYPE_MAP.values():
+        import pdb; pdb.set_trace()
```
Not sure this was meant to make it into the commit?

michaelwood commented on cove/input/models.py:
```diff
@@ -16,8 +18,16 @@
     'text/xml': 'xml',
 }
 
+def original_file_name(file_extension):
```
I'd like to see the original file name still preserved in the string if possible. This improves ux because if you've done quite a number of these and have a downloads folder full of `quai3aenu8Xai5oh.xlsx` files it gets quite difficult to humanly identify them. Suggested change would be ```return f"{filename}-{random_stirng}.{file_extension}" ``` with filename being an argument to `originial_file_name`

jpmckinney replied:
Why not use the `filename` directive of the `Content-Disposition` header to set the filename to the original, in a new view for handling downloads? That way, only the URL is obfuscated, and the filename makes sense to the uploader.

jpmckinney replied:
If a user uploads multiple files and downloads many corresponding files, it'll be very annoying to untease original-garbagetext1 from original-garbagetext2. It'd be a better user experience to just put a view in front of the /media/ files for which the URL is not guessable - while still sending back the filenames as before.

Bjwebb commented:
I've switched to putting the original file in a random subdirectory. This should give randomised urls, and the original file names, whilst allowing us to continue serving the files with the webserver.

Bjwebb committed:
cove.input: Randomize the uploaded file subdirectory

https://github.com/ThreeSixtyGiving/standard/issues/353

This change means that the URL can't be guessed. This gives coves the
option of only giving the URL to authenticated users.

Bjwebb committed:
release: v0.27.0
